### PR TITLE
Simplify GraphActionGNN

### PR DESCRIPTION
**Pull request by younik**
- [x] I've read the [.github/CONTRIBUTING.md](.github/CONTRIBUTING.md) file
- [x] My code follows the typing guidelines
- [ ] I've added appropriate tests
- [x] I've run pre-commit hooks locally

## Description

As a leftover for merging two modules, the GraphActionGNN was using two different heads to compute the logits of the action type. I merged this in a single head, as it makes more sense. I checked the tutorial and they have the same results as before.

### CODE REVIEW

**Comment by josephdviviano** [comment]
Have a linter error:

```  /home/runner/work/torchgfn/torchgfn/tutorials/examples/train_bayesian_structure.py:167:27 - error: Object of type "Tensor" is not callable
```
